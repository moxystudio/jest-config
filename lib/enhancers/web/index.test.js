'use strict';

const withWeb = require('.');

const buildConfig = () => ({
    testEnvironment: 'jsdom',
    transform: {
        foo: 'bar',
    },
    moduleNameMapper: {
        foz: 'baz',
    },
});

it('should match snapshot', () => {
    expect(withWeb()(buildConfig())).toMatchSnapshot();
});

it('should throw if testEnvironment is not jsdom', () => {
    const config = {
        ...buildConfig(),
        testEnvironment: 'node',
    };

    expect(() => withWeb()(config)).toThrow(/expected testEnvironment to be set to "jsdom"/i);
});
