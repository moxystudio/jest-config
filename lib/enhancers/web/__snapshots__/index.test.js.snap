// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Object {
  "moduleNameMapper": Object {
    "/.css$": "<PROJECT_ROOT>/node_modules/identity-obj-proxy/src/index.js",
    "foz": "baz",
  },
  "testEnvironment": "jsdom",
  "transform": Object {
    "/.(eot|ttf|woff|woff2|otf)$": "<PROJECT_ROOT>/lib/transforms/file.js",
    "/.(mp3|flac|wav|aac|ogg|oga|mp4|m4a|webm|ogv)$": "<PROJECT_ROOT>/lib/transforms/file.js",
    "/.(obj|mtl|fnt|glb)$": "<PROJECT_ROOT>/lib/transforms/file.js",
    "/.(png|jpg|jpeg|gif|webp|svg|ico|bmp)$": "<PROJECT_ROOT>/lib/transforms/file.js",
    "/.data-url/.[^/.+?]$": "<PROJECT_ROOT>/lib/transforms/data-url.js",
    "/.inline/.[^/.+?]$": "<PROJECT_ROOT>/lib/transforms/inline.js",
    "foo": "bar",
  },
}
`;
